### ISSUE TITLE
Poor FPS and eventual game freeze.

### ISSUE THREAD

**Issue by anlsh** [ValveSoftware/Dota-2#2273]
System Information
----------------------------------------

* [Steam-reported System Information](https://github.com/anlsh/tech-support/blob/main/march-dota-crashes/steam-sys-info.txt)
* Have you checked for system updates?: Yes
* Are you using the latest stable video driver available for your system? Yes
* Have you verified the game files?: Yes

Issue
-------------------------------
Dota is extremely unstable today (March 10, since 7pm at least). It was totally fine yesterday night (as late as 1am), so I'm guessing an update since then is causing an issue.

First of all, my FPS is very bad. Usually I get around 50 fps. Now I start around 30 or 20, which degrades over the course of a few minutes to 10. Around this point, my game will totally freeze and I will have to kill the process.

This *seems* to be linked to networking somehow. I tried reproing in a bot match and also a private lobby, but without success. However this has forced me out of the two games that I've tried to play today (side note, any chance I can get my abandons rescinded? player ID 108017399), and I can reproduce the issue consistently by trying to watch any game on DotaTV.

I've repro'd the issue twice now, and have uploaded logs to [this repo](https://github.com/anlsh/tech-support/tree/main/march-dota-crashes). I've included output from `dmesg` and `journalctl` (and, for the first crash, a dump file which was mentioned by journalctl).

The `dmesg` logs look particularly suspicious: the nvidia messages line up neatly with my game freezes.

Repro
-----------------------------

1. Start dota, and find a match or start watching a game on dotatv
2. Observe awful and degrading performance and eventual freeze

**Comment by anlsh** [comment]
Just reproduced a third freeze. This time it happened with 10-20 seconds of watching a game. So more evidence that there might be more than one issue here.

I've updated the log repo in the OP: for this one I managed to figure out how to grab dota's `console.log` too. That file seems pretty informative: I see complaints about `NetChan` processing packets slowly, and also something about Vulkan losing a device(!) that I'm guessing is the cause of the freeze

**Comment by paboum** [comment]
Same for me. Looks like overheating GPU, but it's not. Crashes even in main menu. It is most likely related to the opengl support drop.
```
[Mon Mar 13 03:23:36 2023] nvidia-modeset: ERROR: GPU:0: Idling display engine timed out: 0x0000927c:1:0:1128
[Mon Mar 13 03:24:07 2023] steamwebhelper[6757]: segfault at c ip 00007f7f6f19b11d sp 00007ffc20a59b68 error 6 in libnvidia-glcore.so.525.89.02[7f7f6de7f000+2505000]
[Mon Mar 13 03:24:07 2023] steamwebhelper[6763]: segfault at c ip 00007f7f6f19b11d sp 00007ffc20a59b68 error 6 in libnvidia-glcore.so.525.89.02[7f7f6de7f000+2505000]
[Mon Mar 13 03:35:38 2023] nvidia-modeset: ERROR: GPU:0: Idling display engine timed out: 0x0000927c:1:0:1128
[Mon Mar 13 03:40:12 2023] nvidia-modeset: ERROR: GPU:0: Idling display engine timed out: 0x0000927c:1:0:1128
```
No xorg server errors. nvidia-drivers 525.89.02 No logs in `~/.local/share/Steam/steamapps/common/dota 2 beta/` - but it seems the game is still in beta phase.

Interestingly, seems to help when I switch from 4K to HD resolution. But they don't pay me enough to beta-test it any more, I will simply stop purchasing anything until this bug gets fully explained and resolved.

**Comment by anlsh** [comment]
FPS has been fine for the past two days, but dota still crashes (just now with no warning ugh).

Still getting the same errors right around the freezes

```
Mar 14 20:32:15 pop-os kernel: Asynchronous wait on fence NVIDIA:nvidia.prime:12f2fa timed out (hint:intel_atomic_commit_ready [i915])
Mar 14 20:32:17 pop-os /usr/libexec/gdm-x-session[3945]: (EE) NVIDIA(GPU-0): WAIT (1, 8, 0x8000, 0x00020f34, 0x000212f8)
```

Lost the `console.log` because I had to reboot my machine and restart dota in a hurry (twice!) to make sure I didn't abandon my last game

**Comment by anlsh** [comment]
This is still happening. And whatever's going on, I can't even `ctrl + alt + f2` to get to my serial console. Something is seriously wrong :/ here

EDIT: Dug up some new messages from the logs. Unfortunately, it looks like this might be some sort of driver issue https://forums.developer.nvidia.com/t/display-freezes-ee-nvidia-gpu-0-wait/194940
```
Mar 20 20:10:14 pop-os steam[32879]: g_main_context_push_thread_default: assertion 'acquired_context' failed
Mar 20 20:10:14 pop-os steam[32879]: g_main_context_pop_thread_default: assertion 'g_queue_peek_head (stack) == context' failed
Mar 20 20:10:14 pop-os org.kde.kdeconnect.daemon.desktop[4430]: "Object does not exist at path “/org/freedesktop/NetworkManager/ActiveConnection/1”"
Mar 20 20:10:17 pop-os /usr/libexec/gdm-x-session[3970]: (EE) NVIDIA(GPU-0): WAIT (2, 8, 0x8000, 0x00019878, 0x0001a048)
Mar 20 20:10:20 pop-os kernel: Asynchronous wait on fence NVIDIA:nvidia.prime:125e07 timed out (hint:intel_atomic_commit_ready [i915])
Mar 20 20:10:24 pop-os /usr/libexec/gdm-x-session[3970]: (EE) NVIDIA(GPU-0): WAIT (1, 8, 0x8000, 0x00019878, 0x0001a048)
Mar 20 20:10:27 pop-os /usr/libexec/gdm-x-session[3970]: (EE) NVIDIA(GPU-0): WAIT (2, 8, 0x8000, 0x00019878, 0x0001a064)
Mar 20 20:10:34 pop-os /usr/libexec/gdm-x-session[3970]: (EE) NVIDIA(GPU-0): WAIT (1, 8, 0x8000, 0x00019878, 0x0001a064)
Mar 20 20:10:37 pop-os /usr/libexec/gdm-x-session[3970]: (EE) NVIDIA(GPU-0): WAIT (2, 8, 0x8000, 0x00019878, 0x0001a06c)
Mar 20 20:10:44 pop-os /usr/libexec/gdm-x-session[3970]: (EE) NVIDIA(GPU-0): WAIT (1, 8, 0x8000, 0x00019878, 0x0001a06c)
Mar 20 20:10:49 pop-os /usr/libexec/gdm-x-session[3970]: (WW) NVIDIA: Wait for channel idle timed out.
Mar 20 20:10:52 pop-os /usr/libexec/gdm-x-session[3970]: (EE) NVIDIA(GPU-0): WAIT (2, 8, 0x8000, 0x00019878, 0x0001a074)
Mar 20 20:10:59 pop-os /usr/libexec/gdm-x-session[3970]: (EE) NVIDIA(GPU-0): WAIT (1, 8, 0x8000, 0x00019878, 0x0001a074)
Mar 20 20:11:02 pop-os /usr/libexec/gdm-x-session[3970]: (EE) NVIDIA(GPU-0): WAIT (2, 8, 0x8000, 0x00019878, 0x0001a07c)
Mar 20 20:11:09 pop-os /usr/libexec/gdm-x-session[3970]: (EE) NVIDIA(GPU-0): WAIT (1, 8, 0x8000, 0x00019878, 0x0001a07c)
Mar 20 20:11:12 pop-os /usr/libexec/gdm-x-session[3970]: (EE) NVIDIA(GPU-0): WAIT (2, 8, 0x8000, 0x00019878, 0x0001a084)
Mar 20 20:11:19 pop-os /usr/libexec/gdm-x-session[3970]: (EE) NVIDIA(GPU-0): WAIT (1, 8, 0x8000, 0x00019878, 0x0001a084)

```

**Comment by paboum** [comment]
> This is still happening. And whatever's going on, I can't even `ctrl + alt + f2` to get to my serial console. Something is seriously wrong :/ here

Yup, your GPU is in a broken state for some reason. You need to un-jam it by restarting your window manager. Either do what I said in previous comment, or perhaps configure a keyboard shortcut for resetting xorg/wayland. Perhaps a SysRq key might become useful, but I didn't want to spend my time investigating that. Just playing in "just" HD resolution seems as a bearable workaround for me.
> Unfortunately, it looks like this might be some sort of driver issue https://forums.developer.nvidia.com/t/display-freezes-ee-nvidia-gpu-0-wait/194940

Coulda shoulda. I'm using the most recent stable nvidia drivers (525.89.02), this product supports these. It means, even if the drivers are faulty, it's still not my job to worry about it. Valve should have tested their game with these drivers prior to releasing the new version. Maybe Valve should write their own drivers, I don't care.
